iothub/device/src/InternalClient.cs

```diff
@@ -764,52 +753,30 @@ public async Task SetMethodHandlerAsync(string methodName, MethodCallback method
         /// Sets a new delegate for the named method. If a delegate is already associated with
         /// the named method, it will be replaced with the new delegate.
         /// A method handler can be unset by passing a null MethodCallback.
+        /// </summary>
         /// <param name="methodName">The name of the method to associate with the delegate.</param>
         /// <param name="methodHandler">The delegate to be used when a method with the given name is called by the cloud service.</param>
         /// <param name="userContext">generic parameter to be interpreted by the client code.</param>
         /// <param name="cancellationToken">A cancellation token that can be used by other objects or threads to receive notice of cancellation.</param>
-        /// </summary>
         public async Task SetMethodHandlerAsync(string methodName, MethodCallback methodHandler, object userContext, CancellationToken cancellationToken)
         {
             if (Logging.IsEnabled)
-            {
                 Logging.Enter(this, methodName, methodHandler, userContext, nameof(SetMethodHandlerAsync));
-            }
 
             cancellationToken.ThrowIfCancellationRequested();
-            await _methodsDictionarySemaphore.WaitAsync(cancellationToken).ConfigureAwait(false);
+            await _methodsSemaphore.WaitAsync(cancellationToken).ConfigureAwait(false);
 
             try
             {
                 if (methodHandler != null)
                 {
-                    // codes_SRS_DEVICECLIENT_10_005: [ It shall EnableMethodsAsync when called for the first time. ]
-                    await EnableMethodAsync(cancellationToken).ConfigureAwait(false);
-
-                    // codes_SRS_DEVICECLIENT_10_001: [ It shall lazy-initialize the deviceMethods property. ]
```

Comment:
Look at how much simpler the code here gets when we relegate the logic to (the now renamed) `EnableMethodAsync` and `DisableMethodAsync`.